### ISSUE TITLE
Pass correct lineno to Logger?

### ISSUE THREAD

**Issue by cledoux** [kalekundert/nonstdlib#1]
I've been using and loving your `debug.py` module. Thanks for such a great utility!

I have a use question I'm hoping you can help answer: How can I pass the correct line number to the Logger instance for use in the formatted log message?

When debugging, I like to add the line number of the line with the log message for quick code navigation. However, when using debug, the line number is always the line number of the logging call in debug._log.  I know how to retrieve the correct line number from the stack frame, I just don't know how to replace the Logger.lineno attribute.

Any direction you can provide would be greatly appreciated. Thanks!

**Comment by kalekundert** [comment]
I'm glad you like the ``debug.py`` module!  I had no idea anyone but me was actually using this code, so I'm kind of excited to realize that that's not the case!

I looked into how to get the right line numbers a little bit this morning.  I think the easiest way is to monkey-patch the ``findCaller()`` method of the logger object that gets created at the end of ``_log()``.  This is the method that sets the file name, module name, and line number attributes of the logger object.  Of course it uses the current frame, so we just need to write a version that uses the frame object that is being used by the rest of ``_log()``.

I have some free time today, so I'll implement something like this and see how it goes.

**Comment by cledoux** [comment]
Thanks, I appreciate it. I had a method I attempted that made use of the optional extra keyword argument to log, but that broke when I mixed calls to _log and Logger objects obtained via direct calls to getLogger. (I have legacy code that continues to directly call getLogger.)

I've submitted a pull request so you can see the few modifications I've made to the module, if you're interested. The main one is a wrapper around basicConfig that lets me set a stream and file handler all at once.

**Comment by kalekundert** [comment]
Alright, the I just pushed some commits that should fix the line number issue.  It turns out that monkey-patching ``findCaller()`` was going to be hairy, because its return value changed between python2 and python3.  Instead I monkey-patched ``logging.currentframe()``, which was easier and should work seamlessly in any version of python.  

Thanks for the pull request!  I just merged it in.  Feel free to make more pull requests if there are ever any other features you'd like to see in this library.
